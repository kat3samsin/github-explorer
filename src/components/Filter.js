import React, { Component } from 'react';
import { filter, clearFilter } from "../actions/getProjects";
import { connect } from 'react-redux';

class Pagination extends Component {
  applyFilters = () => {
    const filters = {
      type: 'language',
      value: this.getFilter.value.toLowerCase()
    };

    this.props.dispatch(filter(filters));
  }
  clearFilters = () => {
    this.props.dispatch(clearFilter());
    this.getFilter.value = "";
  }
  
  render() {
    return (
      <div className='filterform'>
        <input className='filter' required type="text" ref={(input => this.getFilter = input)} placeholder="Enter language" />&nbsp;
        <button className='filterbutton' onClick={this.applyFilters}>APPLY</button>&nbsp;
        <button className='filterbutton' onClick={this.clearFilters}>CLEAR</button>
      </div>
    );
   }
}

const mapStateToProps = (state) => {
  return {
      results: state
  }
}

export default connect(mapStateToProps)(Pagination);