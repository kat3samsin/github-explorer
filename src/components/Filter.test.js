import React from 'react';
import { shallow } from 'enzyme';
import { Filter } from './Filter';

describe('Filter', () => {
  const applyFiltersFn = jest.fn();
  const clearFiltersFn = jest.fn();
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Filter applyFilters={applyFiltersFn} clearFilters={clearFiltersFn} />)
  });
  afterEach(() => {
    applyFiltersFn.mockReset();
    clearFiltersFn.mockReset();
  })

  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('when applyFilters button is clicked', () => {
    it('should call mock applyFiltersFn', () => {
      wrapper.find('#applyFilters').prop('onClick')({preventDefault() {}});
      expect(applyFiltersFn.mock.calls.length).toBe(1);
    });
  });

  describe('when clearFilters button is clicked', () => {
    it('should call mock clearFiltersFn', () => {
      jest.mock('./Filter');
      Filter.prototype.getFilter = {value: ''};

      wrapper.find('#clearFilters').prop('onClick')({preventDefault() {}});
      expect(clearFiltersFn.mock.calls.length).toBe(1);
    });
  });

  describe('when enter is pressed', () => {
    it('should call mock applyFiltersFn', () => {
      wrapper.find('input').prop('onKeyUp')({keyCode: 13})
      expect(applyFiltersFn.mock.calls.length).toBe(1);
    });

  });

  describe('when a is pressed', () => {
    it('should not call applyFiltersFn', () => {
      wrapper.find('input').prop('onKeyUp')({keyCode: 65})
      expect(applyFiltersFn.mock.calls.length).toBe(0);
    });
  });
});