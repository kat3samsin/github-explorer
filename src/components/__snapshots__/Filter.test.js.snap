// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Filter
    applyFilters={[MockFunction]}
    clearFilters={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="filter"
          onKeyUp={[Function]}
          placeholder="Enter language"
          required={true}
          type="text"
        />,
        " ",
        <button
          className="filterbutton"
          id="applyFilters"
          onClick={[Function]}
        >
          APPLY
        </button>,
        " ",
        <button
          className="filterbutton"
          id="clearFilters"
          onClick={[Function]}
        >
          CLEAR
        </button>,
      ],
      "className": "filterform",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "filter",
          "onKeyUp": [Function],
          "placeholder": "Enter language",
          "required": true,
          "type": "text",
        },
        "ref": [Function],
        "rendered": null,
        "type": "input",
      },
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "APPLY",
          "className": "filterbutton",
          "id": "applyFilters",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "APPLY",
        "type": "button",
      },
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "CLEAR",
          "className": "filterbutton",
          "id": "clearFilters",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "CLEAR",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="filter"
            onKeyUp={[Function]}
            placeholder="Enter language"
            required={true}
            type="text"
          />,
          " ",
          <button
            className="filterbutton"
            id="applyFilters"
            onClick={[Function]}
          >
            APPLY
          </button>,
          " ",
          <button
            className="filterbutton"
            id="clearFilters"
            onClick={[Function]}
          >
            CLEAR
          </button>,
        ],
        "className": "filterform",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "filter",
            "onKeyUp": [Function],
            "placeholder": "Enter language",
            "required": true,
            "type": "text",
          },
          "ref": [Function],
          "rendered": null,
          "type": "input",
        },
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "APPLY",
            "className": "filterbutton",
            "id": "applyFilters",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "APPLY",
          "type": "button",
        },
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "CLEAR",
            "className": "filterbutton",
            "id": "clearFilters",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "CLEAR",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
